extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzBuildTimeConfig.java

```diff
@@ -38,4 +42,29 @@
      */
     @ConfigItem(name = "datasource")
     public Optional<String> dataSourceName;
+
+    /**
+     * The jdbc store class to use when the `db` store type is set.
+     * <p>
+     * There are actually two seperate JDBCJobStore classes {@link org.quartz.impl.jdbcjobstore.JobStoreCMT}
+     * and {@link org.quartz.impl.jdbcjobstore.JobStoreTX} that you can select between, depending on the transactional
+     * behaviour you need.
+     * <p>
+     * The {@link org.quartz.impl.jdbcjobstore.JobStoreCMT} is meant to be used in an application-server
+     * environment that provides container-managed-transactions. No commit / rollback will be handled by this class.
+     * <p>
+     * The {@link org.quartz.impl.jdbcjobstore.JobStoreTX} manages all transactions for you by handling commit and rollback.
+     * <p>
+     * If not specified, defaults to using {@link org.quartz.impl.jdbcjobstore.JobStoreCMT}.
+     */
+    @ConfigItem(defaultValue = "org.quartz.impl.jdbcjobstore.JobStoreCMT")
+    public Optional<String> jdbcStoreClass;
```

Comment:
I wonder if it should be more generic and not tied to jdbc. I could envision some customization even if not jdbc.

Reply:
Since we have only two possible classes I was thinking of an enum:

```
enum TransactionStrategy {
  MANAGED, // this will be JobStoreTX
  UNMANAGED;  // this will be JobStoreCMT
}
```

The we could have a config property named like `quarkus.quartz.transaction-strategy`, I was just not sure if it is the good way to go. Are you thinking of something like that or along those lines?

Reply:
I'm not sure it makes sense to use generic names here. It's Quartz-specific. On the other hand, it could make sense to use the enum. In fact, it could make sense to change the enum for the `QuartzBuildTimeConfig.storeType` to `RAM`, `DB`, `JDBC_TX` and `JDBC_CMT` (where `DB` is an alias of `JDBC_CMT`) or something like that.

Reply:
+1, I like this better. 

Let's go this way and not add another configuration knob.